SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -658,20 +657,29 @@ void USpatialNetDriver::OnLevelAddedToWorld(ULevel* LoadedLevel, UWorld* OwningW
 		return;
 	}
 
-	// If we have authority over the GSM when loading a sublevel, make sure we have authority
-	// over the actors in the sublevel.
-	if (GlobalStateManager != nullptr)
+	check(GlobalStateManager != nullptr)
+
+	// If load balancing disabled but this worker is GSM authoritative then make sure
+	// we set Role_Authority on Actors in the sublevel. Also, if load balancing is
+	// enabled and lb strategy says we should have authority over a loaded level Actor
+	// then also set Role_Authority on Actors in the sublevel.
+	const bool bLoadBalancingEnabled = GetDefault<USpatialGDKSettings>()->bEnableUnrealLoadBalancer;
+	const bool bHaveGSMAuthority = StaticComponentView->HasAuthority(SpatialConstants::INITIAL_GLOBAL_STATE_MANAGER_ENTITY_ID, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID);
```

Comment:
Can `HasAuthority` on the GSM?

Reply:
We removed that, since internally it checked for authority on the SINGLETON_MANAGER_COMPONENT, which is a bit misleading if you want to check authority on other components that the GSM manages.

Reply:
👍